XMonad/Actions/CycleWS.hs

```diff
@@ -262,7 +279,43 @@ wsTypeToPred AnyWS      = return (const True)
 wsTypeToPred (WSTagGroup sep) = do cur <- groupName.workspace.current <$> gets windowset
                                    return $ (cur ==).groupName
                                    where groupName = takeWhile (/=sep).tag
-wsTypeToPred (WSIs p)   = p
+wsTypeToPred (WSIs p ) = p
+wsTypeToPred (p :&: q) = liftM2 (&&) <$> wsTypeToPred p <*> wsTypeToPred q
+wsTypeToPred (p :|: q) = liftM2 (||) <$> wsTypeToPred p <*> wsTypeToPred q
+wsTypeToPred (Not p  ) = fmap not <$> wsTypeToPred p
+
+
+emptyWS, hiddenWS, anyWS :: WSType
+
+-- | cycle through empty workspaces
+emptyWS = WSIs . return $ isNothing . stack
+
+-- | cycle through non-visible workspaces
+hiddenWS = WSIs $ do
+  hs <- gets (map tag . hidden . windowset)
+  return $ (`elem` hs) . tag
+
+-- | cycle through all workspaces
+anyWS = WSIs . return $ const True
+
+-- | cycle through workspaces that are not in the given list. This could be used
+--   ,for example, to skip the workspace used for skipping the workspace reserved
```

Comment:
```suggestion
--   , for example, be used for skipping the workspace reserved
```

Reply:
Hmm, seems perfectly fine to me (jk, I have no idea how I ended up writing that :upside_down_face: )